tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/BaseLinker.scala

```diff
@@ -115,7 +116,18 @@ final class BaseLinker(semantics: Semantics, esLevel: ESLevel, considerPositions
       }
 
       val linkingErrLevel = if (fatal) Level.Error else Level.Warn
-      analysis.errors.foreach(logError(_, logger, linkingErrLevel))
+      val maxDisplayErrors = {
+        val propName = "org.scalajs.core.tools.linker.maxlinkingerrors"
+        Try(System.getProperty(propName, "20").toInt).getOrElse(20).max(1)
```

Comment:
Have you considered adding some option to `logger`? It seems to me that this is really a property of the logger and not of the `BaseLinker`.

Between a `DynamicVariable` and the `SystemProperty` I'd go with the `SystemProperty` as well, because the typing of `DynamicVariable` doesn't buy much but makes configuration much harder.